backend/src/main/java/com/htecgroup/skynest/service/impl/CurrentUserServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package com.htecgroup.skynest.service.impl;
+
+import com.htecgroup.skynest.exception.UserException;
+import com.htecgroup.skynest.exception.UserExceptionType;
+import com.htecgroup.skynest.model.entity.UserEntity;
+import com.htecgroup.skynest.repository.UserRepository;
+import com.htecgroup.skynest.service.CurrentUserService;
+import lombok.AllArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.UUID;
+
+@Service
+@AllArgsConstructor
+public class CurrentUserServiceImpl implements CurrentUserService {
+
+  private UserRepository userRepository;
+
+  @Override
+  public Authentication getAuthentication() {
```

Comment:
Consider the option to return the logged User object, (maybe that will trigger some changes in that model as well, like extending spring security User object) and instead of 3 methods getAuthentication, getEmail and getRole, you will have just one and that will be getLoggedUser()

Reply:
I'll take a look how it can be done.